test/common/upstream/health_checker_impl_test.cc

```diff
@@ -16,21 +15,16 @@
 #include "common/json/json_loader.h"
 #include "common/network/utility.h"
 #include "common/protobuf/utility.h"
-#include "common/upstream/health_checker_impl.h"
 #include "common/upstream/upstream_impl.h"
 
 #include "test/common/http/common.h"
+#include "test/common/upstream/health_checker_impl_test_utils.h"
 #include "test/common/upstream/utility.h"
 #include "test/mocks/access_log/mocks.h"
 #include "test/mocks/api/mocks.h"
-#include "test/mocks/common.h"
-#include "test/mocks/http/mocks.h"
```

Comment:
i think this ends up being directly used in this file, so it's best to include it (MockClientConnection). I believe these headers are also split per mock. Searching for `Http::Mock` willl find which ones are directly needed. (same with `Network::Mock`

Reply:
These are included in .h file from utils class. Should I reinclude??????

Reply:
I guess this maps to a general question I've always had - if it's already included in a header that you include, do you still need to include it?

Reply:
Technically yes, lets say a depends on b and and b brings in c dependencies. Then, "Someone refactors b so that it no longer depends on c, inadvertently breaking a through no fault of their own." See that link https://docs.bazel.build/versions/master/build-ref.html#actual_and_declared_dependencies

Ideally you want to include direct dependencies (and no more) and include those in the build rule. Transitive dependencies can break targets if the middle changes.

Reply:
Sounds good, thanks! Will change now